aarch64-dit/src/lib.rs

```diff
@@ -0,0 +1,64 @@
+#![no_std]
+#![doc = include_str!("../README.md")]
+#![doc(
+    html_logo_url = "https://raw.githubusercontent.com/RustCrypto/meta/master/logo.svg",
+    html_favicon_url = "https://raw.githubusercontent.com/RustCrypto/meta/master/logo.svg"
+)]
+#![warn(missing_docs, rust_2018_idioms, unused_qualifications)]
+
+#[cfg(not(target_arch = "aarch64"))]
+compile_error!("This crate only builds on `aarch64` targets");
+
+use core::arch::asm;
+
+/// Detect if DIT is enabled for the current thread by checking the processor state register.
+#[target_feature(enable = "dit")]
+pub unsafe fn get_dit_enabled() -> bool {
+    let mut dit: u64;
+    asm!("mrs {dit}, DIT", dit = out(reg) dit);
+    (dit >> 24) & 1 != 0
+}
+
+/// Enable DIT for the current thread.
+#[target_feature(enable = "dit")]
+pub unsafe fn set_dit_enabled() {
+    asm!("msr DIT, #1");
```

Comment:
I think it's worth to add proper `options` to this and other `asm!` blocks. I think `nomem`, `preserves_flags`, and `nostack` are applicable.

Reply:
Added in 50a4d9e